### PR TITLE
Integrate `TransformProgram` with `QNode`

### PR DESCRIPTION
**Pull request by albi3ro**
This PR integrates the `TransformProgram` with both the `QNode` and `qml.execute`.

### CODE REVIEW

**Comment by albi3ro** [comment]
[sc-36420]

**Review by timmysilv — COMMENTED** [review]
looks super clean 🎉  just one thought: do we want to advertise using `partial` when decorating a qnode with a transform? I know decorator syntax makes it tough, but I wonder if we can make this better for users without compromising code quality too much

ps: did you mean to delete the specs test (seeing as they're marked to fail)?

**Review by rmoyard — CHANGES_REQUESTED** [review]
Hi @albi3ro, thanks for the PR. The transform rework is also a project to bring us closer to catalyst and eventually to be able to execute a hybrid program `execute(program)`. I would suggest to follow what we accepted in the PRD/ADR and add the program to the execution function.

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -11,6 +11,31 @@
...
+dev = qml.devices.experimental.DefaultQubit2()
+
+@partial(scale_shots, shot_scaling=2)
+@qml.qnode(dev, interface=None)
+def circuit():
+    return qml.sample(wires=0)
+
+```
+
+>>> circuit(shots=1)
```
Comment (rmoyard) [inline doc/releases/changelog-dev.md:36]: Does it use the override shot here? Maybe simpler example with device shot is better for most users

File: pennylane/transforms/core/transform.py [hunk 2]
```diff
@@ -156,7 +156,7 @@ def _transform_signature_check(signature):
             "pennylane.tape.tape.QuantumTape], <built-in function callable>)"
         )
 
-    if not ret[0] in (
+    if ret[0] not in (
```
Comment (rmoyard) [inline pennylane/transforms/core/transform.py:159]: Good catch, thanks

**Review by rmoyard — APPROVED** [review]
Thanks it looks good :+1:

File: doc/releases/changelog-dev.md (hunk 1 above)
Comment (albi3ro) [reply doc/releases/changelog-dev.md:36]: Definitely could use an improved example. But we can have a more polished example put together before the release.

**Pull request merged**